CONTRIBUTING.md

```diff
@@ -1,18 +1,149 @@
-# Contributing
+# Contributing to ICU4X Project
+
+`ICU4X` is an open source project and welcomes everyone to participate.
+
+In order to provide meaningful contributions, it is important to familiarize yourself with a set of documents which describe the [structure](./docs/charter.md) and [style](./docs/style-guide.md) used by the project.
+
+The list of [open issues](triaging.md) represent the current focus points of the project, and the [help wanted](https://github.com/unicode-org/icu4x/issues?q=is%3Aissue+label%3A%22help+wanted%22+no%3Aassignee+) lists the issues which we don't have resources to work on right now, but would consider accepting if someone volunteered to work on them.
+
+Issues are open to everyone to discuss and can be used to jump-start Pull Requests intended for the project.
+
+In most cases, the first step is to find or file a new issue related to your planned contribution, discuss it, and once you received a feedback indicating that the pull request would be welcomed you can start working on it.
+
+## Contributing a Pull Request
+
+The first step is to fork the repository to your namespace and create a branch off of the `master` branch to work with.
+
+That branch may end up containing one of more commits that are constituting the full scope of the pull request.
+
+### Checklist
+
+Each commit and pull request should follow the [style guide](./docs/style-guide.md) and pass all checks such as tests and `cargo fmt`. If the PR is adding any public API changes, we'd also like to ensure that full coverage of `cargo doc` is preserved and code coverage hovers above `90%`.
```

Comment:
optional: "hovers around" instead of "hovers above".   Another alternative, considering that you already have "we'd ... like" = "we would ... like", which already expresses room for lenience, is to just say "is above" or "is at least".